### PR TITLE
[#166761825] Badge to show new services

### PR DESCRIPTION
**Pull request by chiadav**
Add a badge on new services, the badge is present until the user read the service information details.

Android:
android

iPhone 8:
iPhone8

Iphone SE:
iPhoneSE

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#166761825](https://www.pivotaltracker.com/story/show/166761825): Come CIT voglio vedere evidenziati eventuali nuovi servizi

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/components/CustomBadge.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
+ * A component to render the circolar badge
+ * TODO: use the same component on all lists (messages, services, transaction)
```
Comment (fpersico) [inline ts/components/CustomBadge.tsx]: Every TODO must be associated with a pivotal story.

File: ts/store/reducers/entities/services/servicesByReadStatus.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import * as pot from "italia-ts-commons/lib/pot";
+import { getType } from "typesafe-actions";
+import { ServicePublic } from "../../../../../definitions/backend/ServicePublic";
+import { clearCache } from "../../../actions/profile";
+import { showServiceDetails } from "../../../actions/services";
+import { Action } from "../../../actions/types";
+import { GlobalState } from "../../types";
+
+export type ServicesByReadState = Readonly<{
```
Comment (fpersico) [inline ts/store/reducers/entities/services/servicesByReadStatus.ts]: ServicesByOrganizationId mean an object with organizationId as key and services as value.
For the same logic ServicesByReadState mean an object with readState as key and services as value, and it is not your case. Find an appropriate name.
Also is there a reason to store the whole Pot object as value? Isn't "true" enough?

File: ts/components/services/ServiceSectionListComponent.tsx [hunk 3]
```diff
@@ -38,13 +40,33 @@ const styles = StyleSheet.create({
...
+    const service =
+      pot.isLoading(potService) ||
+      pot.isError(potService) ||
+      pot.isNone(potService)
+        ? undefined
+        : potService.value;
+
+    return (
+      readServices !== undefined &&
+      Object.keys(readServices).find(k => {
```
Comment (fpersico) [inline ts/components/services/ServiceSectionListComponent.tsx]: readServices is an object with serviceId as key, you don't need to create an array.

File: ts/components/CustomBadge.tsx (hunk 1 above)
Comment (chiadav) [reply ts/components/CustomBadge.tsx]: Ok

File: ts/store/reducers/entities/services/servicesByReadStatus.ts (hunk 2 above)
Comment (chiadav) [reply ts/store/reducers/entities/services/servicesByReadStatus.ts]: Fix in 164293d

File: ts/components/services/ServiceSectionListComponent.tsx (hunk 3 above)
Comment (chiadav) [reply ts/components/services/ServiceSectionListComponent.tsx]: Fix in 83181dc

File: ts/store/actions/services.ts [hunk 4]
```diff
@@ -32,11 +32,21 @@ export const loadService = createAsyncAction(
   "SERVICE_LOAD_FAILURE"
 )<string, ServicePublic, string>();
 
+//
+// show service detail
+//
+
+export const showServiceDetails = createStandardAction("SERVICE_LOAD_DETAILS")<
```
Comment (fpersico) [inline ts/store/actions/services.ts]:
```suggestion
export const showServiceDetails = createStandardAction("SERVICE_SHOW_DETAILS")<
```

File: ts/store/reducers/entities/services/readStateByService.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { getType } from "typesafe-actions";
```
Comment (fpersico) [inline ts/store/reducers/entities/services/readStateByService.ts]: The file name must be "readStateByServiceId"

Comment (chiadav) [reply ts/store/reducers/entities/services/readStateByService.ts]: Fix in 6f0b05f

**Pull request merged**